nilearn/input_data/nifti_masker.py

```diff
@@ -449,6 +466,7 @@ def transform_single_imgs(self, imgs, confounds=None, copy=True):
             memory=self.memory,
             verbose=self.verbose,
             confounds=confounds,
+            sample_mask=sample_mask,
```

Comment:
I realize that we can instantiate a `NiftiMasker` with a `sample_mask` and then call `transform` with another `sample_mask`. Unless I'm missing something, both will be sent to `filter_and_mask` (one through `params` and the other explicitly). The one provided to `transform` will then override the one provided to the constructor which is the desired behavior, but I'm wondering if we should warn the user that this is the case... I don't have a strong opinion about this. I you think this is too much we can leave it like this.

```python
import numpy as np
from nilearn.input_data import NiftiMasker
from nilearn import datasets

sample_mask = np.array([1,3,5,7])
masker = NiftiMasker(sample_mask=sample_mask)
dataset = datasets.fetch_development_fmri(n_subjects=1)
masker.fit(dataset.func[0])
time_series = masker.transform(dataset.func[0],
                               sample_mask=np.array([2,4]))
time_series.shape
```
```
(2, 32504)
```

Reply:
Thanks for finding this out! I would prefer to remove the capacity to remove supplying sample_mask from Niftimasker

Reply:
After a second thought - 
- `sample_mask` in `NiftiMasker`, nothing from `transform`: use deprecated attribute `NiftiMasker._sample_mask`
- supply both: show warning, use `sample_mask` from transform